### PR TITLE
Move analyzer and fixer tests into FuncionalTests project

### PR DESCRIPTION
**Pull request by joperezr**
In the PR where they were added, all of the Analyzer and Fixer tests were added to the unit test project as opposed to the functional test project. Given there is no real need for that, this PR is moving them to instead be on the functional Unit test project.

### CODE REVIEW

File: src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj [hunk 1]
```diff
@@ -43,6 +44,9 @@
   </ItemGroup>
   <ItemGroup Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">
     <Compile Include="CustomDerivedRegexScenarioTest.cs" />
+    <Compile Include="CSharpCodeFixVerifier`2.cs"/>
+    <Compile Include="UpgradeToRegexGeneratorAnalyzerTests.cs" />
+    <Compile Include="$(CommonTestPath)SourceGenerators\LiveReferencePack.cs" Link="Common\SourceGenerators\LiveReferencePack.cs" />
```
Comment (stephentoub) [inline src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj:49]: Why are these needed?  Why isn't this just like the generator, referencing the public types exposed from the System.Text.RegularExpressions.Generator.csproj being referenced?

Comment (joperezr) [reply src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj:49]: Not necessarily needed, this is just a different approach from what the source generator tests do. This is the pattern that is followed by the LibraryGenerator analyzer tests, which is just a more centralized way of how to reference the whole live-built ref pack form the tests. That said, I'm fine to remove this complexity and use the source generator tests approach instead, I'll do this in a bit.

File: src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj [hunk 2]
```diff
@@ -61,6 +63,9 @@
     <Compile Include="RegexExperiment.cs" />
     <Compile Include="RegexGeneratorHelper.netcoreapp.cs" />
     <Compile Include="$(CommonTestPath)System\Diagnostics\DebuggerAttributes.cs" Link="Common\System\Diagnostics\DebuggerAttributes.cs" />
+
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="$(MicrosoftCodeAnalysisVersion)" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.CodeFix.Testing.XUnit" Version="$(CompilerPlatformTestingVersion)" />
```
Comment (stephentoub) [inline src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj:68]: What are these references and the CSharpCodeFixVerifier`2.cs needed for?

Comment (stephentoub) [reply src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj:68]: Oh, I see, the existing support we have for validating the analyzer isn't sufficient for actually making changes, hence needing the additional helpers and workspace support.

**Review by stephentoub — APPROVED** [review]
Thanks.

File: src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj (hunk 2 above)
Comment (joperezr) [reply src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/System.Text.RegularExpressions.Tests.csproj:68]: Apart from needing to validate the code fixer as  you point out, this verifier also adds the possibility to validate the exact location of where the diagnostic was emitted, which I believe wasn't possible with the existing source generator infra.
